src/printer_tree.c

```diff
@@ -2418,20 +2417,53 @@ tro_modi_get_notifications(struct trt_tree_ctx *tc)
 }
 
 /**
- * @brief Get next yang-data section if exists.
- *
- * Not implemented.
+ * @brief Get next yang-data section if it is possible.
  *
  * @param[in,out] tc is tree context.
+ * @param[in] u is index to the array of extensions (lysc_ext_instance
+ * or struct lysp_ext_instance).
  * @return Section representation if it exists.
  * @return Empty section representation otherwise.
  */
 static struct trt_keyword_stmt
-tro_modi_next_yang_data(struct trt_tree_ctx *tc)
+tro_modi_next_yang_data(struct trt_tree_ctx *tc, LY_ARRAY_COUNT_TYPE u)
 {
-    tc->section = TRD_SECT_YANG_DATA;
-    /* TODO: yang-data is not supported */
-    return TRP_EMPTY_KEYWORD_STMT;
+    assert(tc);
+    const void *node;
+    const char *yang_data_name;
+
+    if (tc->lysc_tree) {
+        struct lysc_ext_instance *exts;
+        struct lysc_ext_substmt *substmts;
+
+        assert(LY_ARRAY_COUNT(tc->pmod->exts) == LY_ARRAY_COUNT(tc->cmod->exts));
```

Comment:
I just wanted to make sure that, for example, two parsed extensions do not result in some way to one compiled extension. Now that I've looked in more detail at the code, I don't believe it can happen. So I'll delete it.